tests/fsharpqa/Source/run.fs

```diff
@@ -0,0 +1,1479 @@
+
+    // #  close STDERR; open STDERR, ">&SAVEERR"; #resore stderr
+    
+    //print @CommandOutput if ($dumpOutput == 1);
+    do if dumpOutput then printfn "%s" CommandOutput
+    
+    // # Test for an assertion failure
+    //if (-e ASSERT_FILE) {
+    return! 
+        if fileExists ASSERT_FILE |> Option.isSome then
+            //print("Failing Test: Assertion detected. Dump Follows:\n");
+            printfn "Failing Test: Assertion detected. Dump Follows:"
+
+            //open ASSERT, ASSERT_FILE or RunExit(TEST_SKIPPED, "Can't open:" . ASSERT_FILE . " : $!\n");
+            //while (<ASSERT>){ print; }
+            //close ASSERT;
+            try
+                ASSERT_FILE
+                |> getfullpath
+                |> System.IO.File.ReadLines
+                |> Seq.iter (printfn "%s")
+                NUnitConf.genericError "Command Unexpectedly Failed with ASSERT \n"
+            with e ->
+                NUnitConf.skip (sprintf "Can't open: %s : %s" ASSERT_FILE (e.Message))
+        else
+            succeed (cmdExitCode, CommandOutput)
+    }
+
+// #############################################################
+// # RunCompilerCommand -- runs a compiler command line, either directly or through external host
+// #
+let RunCompilerCommand cwd envVars useHosted msg (exe, cmd) dumpOutput =
+    //my ($useHosted, $msg, $cmd, $dumpOutput) = @_;
+    ignore "are arguments"
+    
+    //if($useHosted){
+    if useHosted then
+        //my $port = "11000";
+        //if($ENV{HOSTED_COMPILER_PORT} ne ""){
+        //   $port = $ENV{HOSTED_COMPILER_PORT};
+        //}
+        //
+        //my $attempts = 0;
+        //my $remote = undef;
+        //until($remote || ($attempts == 10)) {
+        //    $remote = IO::Socket::INET->new(
+        //                        Proto    => "tcp",
+        //                        PeerAddr => "localhost",
+        //                        PeerPort => $port,
+        //                    ) or sleep(1);
+        //    $attempts++;                            
+        //}
+        //RunExit(TEST_FAIL, "Unable to connect to hosted compiler \n") unless $remote;
+        //
+        //my $currDir = getcwd();
+        //
+        //# send current directory and full command line to the compiler host process
+        //print $remote "$currDir|||$compiler_command";
+        //
+        //# first line of respone is the exit code
+        //my $ExitCode = 0 + <$remote>;
+        //
+        //# remainder of response is output of compiler
+        //@CommandOutput = <$remote>;
+        //
+        //# still some issues with reliability of hosted compiler.
+        //# if compilation unexpectedly fails, try again with standard compiler
+        //if ($ExitCode && ($Type < TEST_SEEK_ERROR)) {
+        //  return RunCommand($msg, $cmd);
+        //}
+        //
+        //return $ExitCode;
+        TODO "useHosted not supported, let's run the command as is"
```

Comment:
the hosted compiler it's used **to speedup tests**, using a single fsc process instead of call multiple fsc process.
I think we can speedup using parallel testing (by directory or by test), that's was a problem with old runner, nunit just use multiple parallel workers.
if the parallel testing (using normal fsc, not a fsc hosted as service) it's fast enough, we dont need hosted. That better because test is more real world, using a normal fsc.
If it's too slow, i'll implement `useHosted`, but i dont think we need it